### PR TITLE
More sql server tests for firewall and service-objective

### PR DESCRIPTION
**Pull request by vishrutshah**
This PR adds basic nose unit tests and recording for
```
az sql server firewall create
az sql server firewall show
az sql server firewall update
az sql server firewall list
az sql server firewall delete
```
```
az sql server service-objective list
az sql server service-objective show
```

The only bug I found was the required parameters were not correctly documents but looks like it's the swaggers fault instead of the cli. 

```
(env) vishrut@visshamac azure-cli (sql-server-firewall-test) $ az sql server firewall create -h

Command
    az sql server firewall create: Create an Azure SQL server firewall rule.

Arguments
    --firewall-rule-name [Required]: The name of the Azure SQL server firewall rule.
    --resource-group -g  [Required]: Name of resource group.
    --server-name        [Required]: The name of the Azure SQL server.
    --end-ip-address               : The end IP address of the Azure SQL server firewall rule. Must
                                     be IPv4 format.
    --start-ip-address             : The start IP address of the Azure SQL server firewall rule.
                                     Must be IPv4 format.

Global Arguments
    --debug                        : Increase logging verbosity to show all debug logs.
    --help -h                      : Show this help message and exit.
    --output -o                    : Output format.  Allowed values: json, jsonc, list, table, tsv.
                                     Default: json.
    --query                        : JMESPath query string. See http://jmespath.org/ for more
                                     information and examples.
    --verbose                      : Increase logging verbosity. Use --debug for full debug logs.
(env) vishrut@visshamac azure-cli (sql-server-firewall-test) $ 
```

The above command has required parameters `start-ip-address` & `end-ip-address`.[ arm-sql/2014-04-01/swagger/firewallRules.json#L182](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2014-04-01/swagger/firewallRules.json#L182)

Do you guys any guidance on how should we inform the team / just open a bug?

@troydai / @derekbekoe Please take a look when you get a chance. Thanks.

### CODE REVIEW

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py [hunk 1]
```diff
@@ -62,3 +62,135 @@ def body(self):
 
         # test list sql server should be 0
         self.cmd('sql server list -g {}'.format(rg), checks=[JMESPathCheck('length(@)', 0)])
+
+class SqlServerFirewallMgmtScenarioTest(ResourceGroupVCRTestBase):
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: two blank lines between class and function definition. 

[Improve your code style](https://github.com/Azure/azure-cli/wiki/Improve-your-code-style)

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py [hunk 2]
```diff
@@ -62,3 +62,135 @@ def body(self):
...
+
+        # test create sql server with minimal required parameters
+        self.cmd('sql server create -g {} --server-name {} -l {} '
+                 '--administrator-login {} --administrator-login-password {}'
+                 .format(rg, self.sql_server_name, loc, user, password), checks=[
+                     JMESPathCheck('name', self.sql_server_name),
+                     JMESPathCheck('resourceGroup', rg),
+                     JMESPathCheck('administratorLogin', user)])
+
+        # test sql server firewall create
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: The test is fairly self-explanatory. I think we can save the comment here.

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py [hunk 3]
```diff
@@ -62,3 +62,135 @@ def body(self):
...
+        self.cmd('sql server firewall delete --firewall-rule-name {} -g {} --server-name {}'
+                 .format(firewall_rule_2, rg, self.sql_server_name), checks=NoneCheck())
+        self.cmd('sql server firewall list -g {} --server-name {}'
+                 .format(rg, self.sql_server_name), checks=[JMESPathCheck('length(@)', 0)])
+
+        # test delete sql server
+        self.cmd('sql server delete -g {} --server-name {}'
+                 .format(rg, self.sql_server_name), checks=NoneCheck())
+
+class SqlServerServiceObjectiveMgmtScenarioTest(ResourceGroupVCRTestBase):
```
Comment (troydai) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: Same as above.

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py (hunk 2 above)
Comment (vishrutshah) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: I agree with you but personally I like comments. Here tests are truly self explanatory so either we can remove all the comments if we want. I am find with any approach. :)

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py (hunk 1 above)
Comment (vishrutshah) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: Fixed

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py (hunk 3 above)
Comment (vishrutshah) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: Corrected.

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py (hunk 2 above)
Comment (troydai) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: There are a few useful ones, if the test statement itself is not that clear. But the comment I noted here is a perfect example of over-commenting. The `sql server firewall create` literally in the next line of code.

This is not a blocking issue. I'll approve this PR anyway.

Comment (vishrutshah) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py]: Noted & agreed :) I'll keep this in mind 👍

**Pull request merged**